src/module_2/analyzing_parquet_files.md

```diff
@@ -0,0 +1,576 @@
+                                                                 right_on='variant_id', how='left')['product_type'].value_counts().reset_index()
+products_sold_by_type.columns = ['product_type', 'sold']
+```
+
+
+```python
+products_abandoned_by_type = df_abandoned_carts.explode('variant_id').merge(df_inventory[['variant_id', 
+                                                                     'product_type']], on='variant_id', 
+                                                       how='left')['product_type'].value_counts().reset_index()
+products_abandoned_by_type.columns = ['product_type', 'abandoned']
+```
+
+
+```python
+sold_vs_abandoned = products_sold_by_type.merge(products_abandoned_by_type, on='product_type', how='left')
+sold_vs_abandoned['abandonment_rate'] = sold_vs_abandoned['abandoned'] / (sold_vs_abandoned['abandoned'] + sold_vs_abandoned['sold'])
+```
+
+
+```python
+df_q2.sort_values('abandonment_rate', ascending=False).head(10)
+```
+
+
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>product_type</th>
+      <th>sold</th>
+      <th>abandoned</th>
+      <th>abandonment_rate</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>57</th>
+      <td>mixed-bundles</td>
+      <td>2</td>
+      <td>3</td>
+      <td>0.600000</td>
+    </tr>
+    <tr>
+      <th>56</th>
+      <td>medicine-treatments</td>
+      <td>27</td>
+      <td>30</td>
+      <td>0.526316</td>
+    </tr>
+    <tr>
+      <th>46</th>
+      <td>household-sundries</td>
+      <td>146</td>
+      <td>116</td>
+      <td>0.442748</td>
+    </tr>
+    <tr>
+      <th>47</th>
+      <td>medicines-treatments</td>
+      <td>97</td>
+      <td>72</td>
+      <td>0.426036</td>
+    </tr>
+    <tr>
+      <th>45</th>
+      <td>superfoods-supplements</td>
+      <td>159</td>
+      <td>107</td>
+      <td>0.402256</td>
+    </tr>
+    <tr>
+      <th>50</th>
+      <td>low-no-alcohol</td>
+      <td>58</td>
+      <td>39</td>
+      <td>0.402062</td>
+    </tr>
+    <tr>
+      <th>36</th>
+      <td>spirits-liqueurs</td>
+      <td>578</td>
+      <td>350</td>
+      <td>0.377155</td>
+    </tr>
+    <tr>
+      <th>43</th>
+      <td>cider</td>
+      <td>215</td>
+      <td>114</td>
+      <td>0.346505</td>
+    </tr>
+    <tr>
+      <th>24</th>
+      <td>washing-capsules</td>
+      <td>1127</td>
+      <td>581</td>
+      <td>0.340164</td>
+    </tr>
+    <tr>
+      <th>52</th>
+      <td>maternity</td>
+      <td>40</td>
+      <td>20</td>
+      <td>0.333333</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+
+```python
+df_q2.sort_values('abandonment_rate', ascending=False).tail(10)
```

Comment:
I think a histogram of the abandonment_rate would have been interesting to check the distribution of the values. With the histogram (or a kdeplot) you can easily see if there are for example, two separated clusters of abandoned products (or maybe there are no clusters since the histogram is continuous).